legacy-modes/src/index.ts

```diff
@@ -144,49 +144,49 @@ class StateCache<S> {
 
 type Config = {
   sleepTime?: number,
-  maxWorkTime?: number
+  maxWorkTime?: number,
+  mode: Mode<any>
 }
 
-export function legacyMode<S>(mode: Mode<S>, config: Config = {}) {
-  const {sleepTime = 100, maxWorkTime = 100} = config
-  const field = new StateField<StateCache<S>>({
+export const legacyMode = Extension.define<Config>(config => {
+  let field = new StateField<StateCache<any>>({
     init(state: EditorState) { return new StateCache([], 0, null) },
     apply(tr, cache) { return cache.apply(tr) },
-    debugName: "mode"
+    name: "mode"
   })
-
-  let plugin = new Plugin({
-    state: field,
-    view(v: EditorView) {
-      let decorations = Decoration.none, from = -1, to = -1
-      function update(v: EditorView, force: boolean) {
-        let vp = v.viewport
-        if (force || vp.from < from || vp.to > to) {
-          ;({from, to} = vp)
-          const stateCache = v.state.getField(field)!
-          decorations = Decoration.set(stateCache.getDecorations(v.state, from, to, mode))
-          stateCache.advanceFrontier(v.state, from, mode, sleepTime, maxWorkTime).then(() => {
-            update(v, true)
-            v.decorationUpdate()
-          }, () => {})
-        }
-      }
-      return {
-        get decorations() { return decorations },
-        updateViewport: update,
-        updateState: (v: EditorView, p: EditorState, trs: Transaction[]) => update(v, trs.some(tr => tr.docChanged))
+  return [
+    Behavior.stateField.use(field),
+    viewPlugin.use(getViewPlugin(field, config)),
+    Behavior.indentation.use((state: EditorState, pos: number): number => {
+      if (!config.mode.indent) return -1
```

Comment:
@marijnh Many places use magic numbers.
Wouldn't it be better to make it a constant to reveal the intent of the value?